Project collaboration mode docs (#3360)
Unleash/unleash: website/docs/reference/project-collaboration-mode.md

---
title: Project collaboration mode
---

:::info Availability

The change requests feature is an enterprise-only feature that was introduced in **Unleash 4.22.0**.

:::

Unleash project collaboration mode is a feature that allows to specify who can submit change requests. Currently, there are two modes:
- **open** where everyone can submit change requests. This is a default setting for the existing and new Unleash projects.

> - **protected** where admins, editors and project members can submit change requests. Viewers who are not added to projects will not be able to submit change requests.

Author: Yes, fixing it

---
title: Project collaboration mode
---

:::info Availability

> The change requests feature is an enterprise-only feature that was introduced in **Unleash 4.22.0**.

Author: Good catch. Definitely a typo

:::

Unleash project collaboration mode is a feature that allows to specify who can submit [change requests](change-requests.md). Currently, there are two modes:
- **open** where everyone can submit change requests. This is a default setting for the existing and new Unleash projects.

> - **protected** where [admins, project members and owners](rbac.md#standard-roles) can submit change requests. Users who are not added to projects will not be able to submit change requests.

Reviewer: Actually, maybe we could change the wording here. Project members and owners suggest that only these roles can submit change requests. That is not true in the implementation, during my testing I tried giving my viewer a custom role with the least privileges possible and I was allowed to submit CRs. There's a difference between being a project member (holding any role in the project) and having the project role of member / owner in the project.

Author: Do you suggest just saying project members?

Reviewer: Yeah, I think that's an ok terminology. Maybe we also define it in this context, since we have the project member role. Something like `project members (any user holding a role in the project)`

Author: I thought that the following sentence "Users who are not added to projects will not be able to submit change requests." was clarifying this. WDYT?

> - **protected** where [admins, project members and owners](rbac.md#standard-roles) can submit change requests. Users who are not added to projects will not be able to submit change requests.

Reviewer 2: Oh, for context: maybe we can explode this out into two separate sections. That makes it easier to explain each mode in more depth (using multiple paragraphs)

> - **protected** where [admins, project members and owners](rbac.md#standard-roles) can submit change requests. Users who are not added to projects will not be able to submit change requests.

Author: Makes sense. Picking and applying your changes. With one minor change. Instead of "Only [admins, project members and owners]" we'll say "only admins and project members"

> ## Project collaboration mode setting
> 
> Project collaboration mode is controlled from the "Create Project" and "Edit Project" screens.
> 
> ![Create project with collaboration mode](/img/collaboration-mode.png)
> 
> When project collaboration mode is protected all users who are not project members or admins will have all the
> change request operations disabled.
> 
> ## Enabling protected mode for an existing project
> 
> To change the project collaboration mode for an existing project you have to edit the project.

Reviewer 2: This feels like it's veering into how-to territory, but maybe we can rephrase it a bit to keep it as a reference guide.

Also, what happens if you change a project's collaboration mode after a viewer submits a CR? Does it stay? And what about editing it?

```suggestion
## Project collaboration mode setting

You can set a project's collaboration mode when you create a project and at any point after creation.

![Project creation form with a collaboration mode field and corresponding explanation.](/img/collaboration-mode.png)

### Change requests 

When you change a project's collaboration mode from open to protected, users who do not have the right permissions will lose the ability to create new change requests. 

However, existing change requests created by these users will not be deleted. Any users with open change requests will still be able to edit the change request, regardless of their current permissions in the project.
```

Author: So when you make the project protected then the permission checker kicks in and you can only cancel your change request from this moment.

Reviewer 2: So you can cancel or have it approved & applied. It cannot be updated?

Author: Correct. I updated the text in docs to reflect it.

> ![Edit project](/img/edit-project.png)

Reviewer 2: Right, because this is one of the few settings that are available from "edit project" instead of "project settings"? What's the reason for that? And why do we have those two separate sections anyway? 

I've added a suggestion for this link higher up, so I'll add one to remove this line too.

```suggestion
```

> ![Edit project](/img/edit-project.png)

Author: We wanted to delivery this capability very soon and it was the quickest and deliberate route. Eventually we'll create project settings in the next quarter.

---
title: Project collaboration mode
---

:::info Availability

The project collaboration mode is an enterprise-only feature that was introduced in **Unleash 4.22.0**.

:::

A project's **collaboration mode** specifies **who can submit [change requests](change-requests.md)**. There are two collaboration modes:
- [**open**](#open-collaboration-mode)
- [**protected**](#protected-collaboration-mode)

## **Open** collaboration mode

Anyone can submit change requests in an **open** project, regardless of their permissions within the project and globally. This is the default collaboration mode for projects.

The open collaboration mode is the default in Unleash and is how all projects worked in Unleash before the introduction of collaboration modes.

## **Protected** collaboration mode

Only admins and project members can submit change requests in a **protected** project. Other users can not submit change requests.

### Change requests

When you change a project's collaboration mode from open to protected, users who do not have the right permissions will lose the ability to create new change requests.

> However, existing change requests created by these users will not be deleted. Any users with open change requests will still be able to cancel the change request.

Reviewer 2: I'd like to be explicit about these users not being able to update the change request.

```suggestion
However, existing change requests created by these users will not be deleted. Any users with open change requests will still be able to cancel the change request, but they will **not** be able to update the change request.
```

## Project collaboration mode setting

You can set a project's collaboration mode when you create a project and at any point after creation.

![Project creation form with a collaboration mode field and corresponding explanation.](/img/collaboration-mode.png)

> 

Reviewer 2: Mayyyybe something like this?

```suggestion
Because the project's collaboration mode is a core part of the project, the controls to change it are available on the "edit project" page.

![The project-level header section with the "edit project" button highlighted.](/img/edit-project.png)

```
